dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunction.cs

```diff
@@ -432,7 +433,12 @@ await using (enumerator.ConfigureAwait(false))
         // visible to a consumer if that's needed.
         if (ex is OperationCanceledException cancelEx)
         {
-            throw new KernelFunctionCanceledException(kernel, kernelFunction, arguments, result, cancelEx);
+            var kernelEx = new KernelFunctionCanceledException(kernel, kernelFunction, arguments, result, cancelEx);
+            foreach (DictionaryEntry entry in cancelEx.Data)
+            {
+                kernelEx.Data.Add(entry.Key, entry.Value);
```

Comment:
Taking into account that we pass `cancelEx` with populated `Data` as inner exception to `KernelFunctionCanceledException` constructor, do we need to copy `OperationCanceledException.Data` to `KernelFunctionCanceledException.Data`? Because without copying, the `Data` still should be accessible by users using `KernelFunctionCanceledException.InnerException.Data`.

Reply:
I considered that but decided it was more convenient for developers if we promoted the properties to the parent exception.

Reply:
Reverting this change